master/buildbot/reporters/stash.py

```diff
@@ -61,8 +67,22 @@ def send(self, build):
         else:
             status = STASH_INPROGRESS
             description = self.startDescription
+
+        # got_revision could be a string, a dictionary or None
+        got_revision = props.getProperty('got_revision', None)
         for sourcestamp in build['buildset']['sourcestamps']:
             sha = sourcestamp['revision']
+
+            if sha is None:
+                if isinstance(got_revision, dict):
```

Comment:
I thought that before, unfortunately, the revision is not available in the payload on the pull request event, it is available the base branch and the compare branch revision but not the merge commit.
I saw that it is possible to get that commit revision by running git ls-remote {repository} refs/pull-requests/{PR}/merge, but this added many other problems (like needing authentication), what value would you suggest using for revision?